docs/guides/chip_tool_guide.md

```diff
@@ -0,0 +1,808 @@
+
+<a name="using"></a>
+
+## Using CHIP Tool for Matter device testing
+
+This section describes how to use CHIP Tool to test the Matter device. The
+following steps depend on the application clusters that you implemented on the
+device.
+
+This tutorial is using the
+[Matter lighting app example](https://github.com/project-chip/connectedhomeip/tree/master/examples/lighting-app)
+with the Bluetooth LE commissioning. You can use other Matter examples and still
+follow this procedure. If you use a different example, the
+[Step 7](#step-7-control-application-data-model-clusters) may vary depending on
+the clusters implemented in your application.
+
+### Step 1: Prepare the Matter device
+
+Build and program the device with the Matter device firmware by following the
+[Matter lighting-app example](https://github.com/project-chip/connectedhomeip/tree/master/examples/lighting-app)
+documentation.
+
+### Step 2: Enable Bluetooth LE advertising on Matter device
+
+Some examples are configured to advertise automatically on boot. Other examples
+require physical trigger, for example pushing a button. Follow the documentation
+of the Matter device example to learn how Bluetooth LE advertising is enabled
+for the given example.
+
+### Step 3: Make sure the network is set up
+
+To follow the next steps, the Thread or Wi-Fi network must be up and running.
+For instance, the Thread network can be established using
+[OpenThread Border Router](https://openthread.io/codelabs/openthread-border-router#0).
+
+### Step 4: Determine network pairing credentials
+
+You must provide the CHIP Tool with network credentials that will be used in the
+device commissioning procedure to configure the device with a network interface,
+such as Thread or Wi-Fi.
+
+The Matter specification does not define how the Thread or Wi-Fi credentials are
+to be obtained by controller.
+
+#### Thread network credentials
+
+Fetch and store the current Active Operational Dataset from the Thread Border
+Router. This step may vary depending on the Thread Border Router implementation.
+
+If you are using
+[OpenThread Border Router](https://openthread.io/codelabs/openthread-border-router#0)
+(OTBR), retrieve this information using one of the following commands:
+
+-   For OTBR running in Docker:
+
+    ```
+    sudo docker exec -it otbr sh -c "sudo ot-ctl dataset active -x"
+    0e080000000000010000000300001335060004001fffe002084fe76e9a8b5edaf50708fde46f999f0698e20510d47f5027a414ffeebaefa92285cc84fa030f4f70656e5468726561642d653439630102e49c0410b92f8c7fbb4f9f3e08492ee3915fbd2f0c0402a0fff8
+    Done
+    ```
+
+-   For OTBR native installation:
+
+    ```
+    sudo ot-ctl dataset active -x
+    0e080000000000010000000300001335060004001fffe002084fe76e9a8b5edaf50708fde46f999f0698e20510d47f5027a414ffeebaefa92285cc84fa030f4f70656e5468726561642d653439630102e49c0410b92f8c7fbb4f9f3e08492ee3915fbd2f0c0402a0fff8
+    Done
+    ```
+
+For Thread, you might also use a different out-of-band method to fetch the
```

Comment:
@Damian-Nordic, @doublemis1 is this clear enough?

Reply:
Fine by me :)